backend/store/principal.go

```diff
@@ -69,10 +72,14 @@ func composePrincipal(user *UserMessage) (*api.Principal, error) {
 
 // FindUserMessage is the message for finding users.
 type FindUserMessage struct {
-	ID          *int
-	Email       *string
-	Role        *api.Role
-	ShowDeleted bool
+	ID                 *int
+	Email              *string
+	Role               *api.Role
+	ShowDeleted        bool
+	IdentityProviderID *int
```

Comment:
Shall we use the IDP resource ID instead of the integer UID so that we are more consistent with API inputs?